### DISCUSSION TITLE
Is 'vm.power -off' a graceful shutdown , without -force ?

### DISCUSSION THREAD

**Discussion by rakaupuv** [vmware/govmomi#3200] [Community Help]
I've a reason to believe that the VM was abruptly shutdown for an OVA export process to begin. Before I dig into the /var/log to check about the shutdown, I'd like to know if any of you are aware about the `vm.power -off` command behavior. I tried looking into the vm/power.go,  vcsa/shutdown/poweroff.go, and the vapi/appliance/shutdown/shutdown.go, but it stops at the rest call and I couldn't figure out if it passed any override command to shutdown.
The reason is that, when the ova was brought up, some of the log files that some of our services use were corrupted and the last modified timestamp was just before shutdown. It very well could be some other issue but, I just want to rule out this possiblity.
The govc version used is: 0.22.1

Thanks!

**Comment by dougm** [comment]
The `-off` flag is not graceful, use `vm.power -s` for graceful shutdown

**Reply by rakaupuv** [reply]
Thanks. This seems to be what I was looking for but I was told that the VMs can't be gracefully shutdown without VMware tools installed in it. Is it true ?
I tried the `govc vm.power -s <vm-name>` just now and it worked one time and threw the error next several times.
"ServerFaultCode: Cannot complete operation because VMware Tools is not running in this virtual machine."
Any advice/help ?

**Reply by dougm** [reply]
That's right, graceful shutdown requires VMware tools to be installed and running
